src/TorchVision/AutoAugment.cs

```diff
@@ -0,0 +1,172 @@
+                            fill: fill?.FirstOrDefault());
+                    case opType.Rotate:
+                        return F.rotate(img, magnitude, interpolation, fill: fill);
+                    case opType.Brightness:
+                        return F.adjust_brightness(img, 1.0 + magnitude);
+                    case opType.Color:
+                        return F.adjust_saturation(img, 1.0 + magnitude);
+                    case opType.Contrast:
+                        return F.adjust_contrast(img, 1.0 + magnitude);
+                    case opType.Sharpness:
+                        return F.adjust_sharpness(img, 1.0 + magnitude);
+                    case opType.Posterize:
+                        return F.posterize(img, (int)magnitude);
+                    case opType.Solarize:
+                        return F.solarize(img, magnitude);
+                    case opType.AutoContrast:
+                        return F.autocontrast(img);
+                    case opType.Equalize:
+                        return F.equalize(img);
+                    case opType.Invert:
+                        return F.invert(img);
+                    case opType.Identity:
+                        return img; // Pass
+                    default:
+                        throw new ArgumentException($"The provided operator {op_name} is not recognized.");
+                }
+            }
+        }
+
+        internal class RandAugment : AutoAugmentBase, ITransform
+        {
+            public RandAugment(
+                int num_ops = 2,
+                int magnitude = 9,
+                int num_magnitude_bins = 31,
+                InterpolationMode interpolation = InterpolationMode.Nearest,
+                IList<float>? fill = null)
+            {
+                this.num_ops = num_ops;
+                this.magnitude = magnitude;
+                this.num_magnitude_bins = num_magnitude_bins;
+                this.interpolation = interpolation;
+                this.fill = fill;
+            }
+
+            public Tensor call(Tensor img)
+            {
+                var fill = this.fill;
+                var cDim = img.Dimensions - 3;
+                var height = img.shape[cDim + 1];
+                var width = img.shape[cDim + 2];
+                var op_meta = augmentation_space(num_magnitude_bins, (height, width));
+                for (int i = 0; i < num_ops; ++i) {
+                    var op_index = torch.randint(op_meta.Count, new[] { 1 }).ToInt32();
+                    var op_name = op_meta.Keys.ToList()[op_index];
+                    var (magnitudes, signed) = op_meta[op_name];
+                    var magnitude = magnitudes.Dimensions > 0 ? magnitudes[this.magnitude] : 0.0f;
+
+                    if (signed && torch.randint(2, new[] { 1 }).ToBoolean())
+                        magnitude *= -1.0;
+
+                    img = apply_op(img, op_name, magnitude.ToSingle(), interpolation, fill);
```

Comment:
Since you have to have it as a float, anyway (should it be a double??), isn't it better to do the `ToSingle()` call in the true-branch of the conditional expression on line 122, and leave the local variable a float instead of a Tensor?

Reply:
Yeah, I'll make it double too.